### ISSUE TITLE
File folders: Move between global and user-specific

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#12287]
JabRef offers multiple file directories. 

![image](https://user-images.githubusercontent.com/1366654/170296053-3edb4b00-25d8-4231-b007-987a4f2e6ce8.png)

- a) There is no indication in which folder "type" a file recides
- b) It is not possible to move between these folders

The UI just shows "file directory"

![image](https://user-images.githubusercontent.com/1366654/170295845-646f9ecf-dc77-4dcf-a50d-23d4fd6fb823.png)

It should be possible to move from the user-specific file directory to the general file directory (and vice versa)

## Proposal

- UI should NOT state "file directory", but "General file directory" and "User-specific file directory". If a file is already contained in one of these directories, the option should be "Move file to user-specific file directory" or "General file directory" respectively.
- The file icon should be "world + pdf" if stored in the "General file directory"
- The file icon should be "person+ pdf" if stored in the "user-specific file directory"

## Notes

- It is not clear whether the icons work as desribed. Other solutions should be tried. As last resort, the icon color can be used. If the PDF is stored in "General file directory" , it is full black. If not, it is a light gray (as currently)

**Comment by ganesh-vk** [comment]
If I've understood this correctly the solution would be to change the implementation of `moveToDefaultDirectory` in `LinkedFileViewModel` to move to the Library specific file directory if the file is in the User specific file directory, move to the Bib file directory if in Library file directory and User file directory if in bib file directory. So essentially a circular loop following the same order of preferences used throughout jabref. Let me know if this is the right approach. 

One more question:
If the User directory and the Bib directory is configured, do we provide an option to move to Bib dir if the file is currently in User dir?

**Comment by koppor** [comment]
> One more question:\nIf the User directory and the Bib directory is configured, do we provide an option to move to Bib dir if the file is currently in User dir?

Yes. Draw a mermaidjs UML flow chart to show the logic. Add it to docs/equirements/, add some text and an id, link it from the implementation.

Update: https://github.com/JabRef/jabref/blob/main/docs%2Frequirements%2Ffiles.md - change existing headings one level deeper.
